addons/stock/models/stock_move_line.py

```diff
@@ -790,6 +795,35 @@ def get_aggregated_properties(move_line=False, move=False):
 
         return aggregated_move_lines
 
+    def _get_empty_moves(self):
+        backorders = self.env['stock.picking']
+        pickings = self.picking_id
+        while pickings.backorder_ids:
+            backorders |= pickings.backorder_ids
+            pickings = pickings.backorder_ids
+        pickings = (self.picking_id | backorders)
+        aggregated_move_lines = {}
+        for empty_move in pickings.move_lines.filtered(
```

Comment:
Please prefer putting the `filtered` as first condition of the loop. It's better regarding performance (you go through the recordset only once) and in this case more readable
```python
# for item in self.filtered(foo):

for item in self:
    if not foo: 
        continue
```

Reply:
Good comment, you can also apply it in in the `_get_aggregated_product_quantities` loop (line 777) :sweat_smile: